Source/CesiumRuntime/Private/Cesium3DTileset.cpp

```diff
@@ -782,102 +785,164 @@ void ACesium3DTileset::MarkTilesetDirty() {
   UE_LOG(LogCesium, Verbose, TEXT("Tileset marked Dirty"));
 }
 
-std::optional<ACesium3DTileset::UnrealCameraParameters>
-ACesium3DTileset::GetCamera() const {
-  std::optional<UnrealCameraParameters> camera = this->GetPlayerCamera();
+std::vector<ACesium3DTileset::UnrealCameraParameters>
+ACesium3DTileset::GetCameras() const {
+  std::vector<UnrealCameraParameters> cameras = this->GetPlayerCameras();
 
 #if WITH_EDITOR
-  if (!camera) {
-    camera = this->GetEditorCamera();
-  }
+  std::vector<UnrealCameraParameters> editorCameras = this->GetEditorCameras();
+  cameras.insert(
+      cameras.end(),
+      std::make_move_iterator(editorCameras.begin()),
+      std::make_move_iterator(editorCameras.end()));
 #endif
 
-  return camera;
+  return cameras;
 }
 
-std::optional<ACesium3DTileset::UnrealCameraParameters>
-ACesium3DTileset::GetPlayerCamera() const {
+std::vector<ACesium3DTileset::UnrealCameraParameters>
+ACesium3DTileset::GetPlayerCameras() const {
   UWorld* pWorld = this->GetWorld();
   if (!pWorld) {
-    return std::optional<UnrealCameraParameters>();
-  }
-
-  APlayerController* pPlayerController = pWorld->GetFirstPlayerController();
-  if (!pPlayerController) {
-    return std::optional<UnrealCameraParameters>();
+    return {};
   }
 
-  APlayerCameraManager* pCameraManager = pPlayerController->PlayerCameraManager;
-  if (!pCameraManager) {
-    return std::optional<UnrealCameraParameters>();
+  float worldToMeters = 0.01f;
+  AWorldSettings* pWorldSettings = pWorld->GetWorldSettings();
+  if (pWorldSettings) {
+    worldToMeters = pWorldSettings->WorldToMeters;
   }
 
   UGameViewportClient* pViewport = pWorld->GetGameViewport();
   if (!pViewport) {
-    return std::optional<UnrealCameraParameters>();
+    return {};
   }
 
-  FVector location;
-  FRotator rotation;
-  pCameraManager->GetCameraViewPoint(location, rotation);
-  double fov = pCameraManager->GetFOVAngle();
-
   FVector2D size;
   pViewport->GetViewportSize(size);
-
   if (size.X < 1.0 || size.Y < 1.0) {
-    return std::nullopt;
+    return {};
+  }
+
+  TSharedPtr<IStereoRendering, ESPMode::ThreadSafe> pStereoRendering = nullptr;
+  if (GEngine) {
+    pStereoRendering = GEngine->StereoRenderingDevice;
   }
 
-  return UnrealCameraParameters{size, location, rotation, fov};
+  std::vector<ACesium3DTileset::UnrealCameraParameters> cameras;
```

Comment:
pWorld has GetNumControllers() and GetNumPlayerControllers(), so they can use to reserve the size of cameras vector to minimize allocation